### PR TITLE
storage: use session path style in HeadBucket call

### LINKED ISSUE #314: s3 virtual host access resolution inconsistent
Hi,

I've read through the release notes and can see that in v1 it states: Virtual host style bucket name resolving is enabled by default for S3 and GCSS. However this doesn't seem to be the case.

I've created a new bucket miles-bucket-us-east-1

s3://miles-bucket-us-east-1/ would translate to https://miles-test-us-east-1.s3.amazonaws.com however from my network traffic logs I can see that the traffic defaults to https://s3.amazonaws.com/miles-bucket-us-east-1

Command to replicate:
``` 
s5cmd ls s3://miles-bucket-us-east-1/
```

I can't see any options that I need to specify to force the s5cmd to use virtual host. Am I doing something wrong?

### PR DESCRIPTION
**Pull request by seruman**
Resolves #314

### CODE REVIEW

File: storage/s3.go [hunk 1]
```diff
@@ -835,6 +835,7 @@ func setSessionRegion(ctx context.Context, sess *session.Session, bucket string)
 
 	// auto-detection
 	region, err := s3manager.GetBucketRegion(ctx, sess, bucket, "", func(r *request.Request) {
+		r.Config.S3ForcePathStyle = sess.Config.S3ForcePathStyle
```
Comment (igungor) [inline storage/s3.go:838]: could you add a comment about why this change is necessary?

Comment (igungor) [reply storage/s3.go:841]: thanks.

**Review by igungor — APPROVED** [review]

**Comment by igungor** [comment]
@seruman could you also update the changelog please?

**Comment by seruman** [comment]
> @seruman could you also update the changelog please?

Updated.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,7 @@
 - Fixed a bug about precedence of region detection, which auto region detection would always override region defined in environment or profile. ([#325](https://github.com/peak/s5cmd/issues/325))
 - Fixed a bug where errors did not result a non-zero exit code. ([#304](https://github.com/peak/s5cmd/issues/304))
 - Print error if the commands file of `run` command is not accessible. ([#410](https://github.com/peak/s5cmd/pull/410))
+- Updated region detection call to use current session's path style config as it is set to 'Path' style by default. ([#314](https://github.com/peak/s5cmd/issues/314))
```
Comment (igungor) [inline CHANGELOG.md:14]:
```suggestion
- Updated region detection call to use current session's address resolving method ([#314](https://github.com/peak/s5cmd/issues/314))
```

Comment (seruman) [reply CHANGELOG.md]: That is much better, thanks.

**Review by aykutfarsak — APPROVED** [review]

**Pull request merged**